### PR TITLE
Delete AppDomain::IsDefaultDomain

### PR DESCRIPTION
**Pull request by jkotas**

### CODE REVIEW

File: src/coreclr/src/debug/di/process.cpp [hunk 1]
```diff
@@ -8763,18 +8763,9 @@ CordbAppDomain * CordbProcess::CacheAppDomain(VMPTR_AppDomain vmAppDomain)
...
-
-    if (fIsDefaultDomain)
-    {
-        // If this assert fires, then it likely means the target is corrupted.
-        TargetConsistencyCheck(m_pDefaultAppDomain == NULL);
-        m_pDefaultAppDomain = pAppDomain;
-    }
+    // If this assert fires, then it likely means the target is corrupted.
+    TargetConsistencyCheck(m_pDefaultAppDomain == NULL);
+    m_pDefaultAppDomain = pAppDomain;
```
Comment (danmosemsft) [inline src/coreclr/src/debug/di/process.cpp]: `m_pDefaultAppDomain` could be renamed?

**Review by danmoseley — APPROVED** [review]

File: src/coreclr/src/debug/di/process.cpp (hunk 1 above)
Comment (jkotas) [reply src/coreclr/src/debug/di/process.cpp]: I think this name is ok. Many other names (including APIs) and comments talk about default appdomain too.

File: src/coreclr/src/debug/ee/debugger.cpp [hunk 2]
```diff
@@ -9463,15 +9463,6 @@ void Debugger::SendExitAppDomainEvent(AppDomain* pRuntimeAppDomain)
 
     if (CORDebuggerAttached())
     {
-        if (pRuntimeAppDomain->IsDefaultDomain() )
-        {
-            // The Debugger expects to never get an unload event for the default Domain.
-            // Currently we should never get here because g_fProcessDetach will be true by
-            // the time this method is called.  However, we'd like to know if this ever changes
-            _ASSERTE(!"Trying to deliver notification of unload for default domain" );
```
Comment (elinor-fung) [inline src/coreclr/src/debug/ee/debugger.cpp]: Isn't this the part we want to keep, since `pRuntimeAppDomain->IsDefaultDomain()` should always be true?

Comment (danmosemsft) [reply src/coreclr/src/debug/ee/debugger.cpp]: ack - I read it backwards - I think you're right @elinor-fung

Comment (jkotas) [reply src/coreclr/src/debug/ee/debugger.cpp]: This whole function is unreachable codepath. Delete it.

**Pull request merged**